### PR TITLE
ENH add matcher using scipy

### PR DESCRIPTION
**Pull request by beckermr**
This PR adds the matcher from scipy and updates the tests.

### CODE REVIEW

**Comment by beckermr** [comment]
@esheldon @erykoff This one is ready for review!

**Comment by beckermr** [comment]
cc @erykoff

File: smatch/matcher.py [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+            Returned if return_indices is True.
+        i2 : array-like, int
+            Array of indices for query lon/lat.
+            Returned if return_indices is True.
+        distance : array-like, float
+            Array of distance (degrees) for each match pair.
+            Returned if return_indices is True.
+        """
+        coords = hp.rotator.dir2vec(lon, lat, lonlat=True).T
+        # The second tree in the match does not need to be balanced, and
```
Comment (erykoff) [inline smatch/matcher.py:208]: I think this comment should probably be moved up top ... neither tree needs to be balanced as it turns out.

**Review by erykoff — COMMENTED** [review]
Is the only use of esutil in smatch just for the spherical distance computations?  That seems excessive, we could probably just do the math ourselves.

**Comment by beckermr** [comment]
By the same token, we only need healpy for getting unit vectors from the positions. That is easily done ourselves too.

File: smatch/matcher.py (hunk 1 above)
Comment (beckermr) [reply smatch/matcher.py:208]: done.

**Comment by beckermr** [comment]
ok @erykoff This is ready for another look!

**Comment by kadrlica** [comment]
Thanks @beckermr and @erykoff, this looks awesome! I totally agree with pulling out the `healpy` and `esutil` dependencies.

I know that @erykoff doesn't like to let `scipy` spawn its own subprocesses, but I just wanted to call attention to the fact that there is a `workers` (formerly `n_jobs`) argument to [`cKDTree.query`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.cKDTree.query.html) that we could expose to the user if anyone feels strongly about it. I know that my big matching jobs are always limited by memory, so I have been interested in see if this actually provides a significant speed-up (I haven't had a chance to do much testing).

**Pull request merged**